packages/ember-views/lib/mixins/text_support.js

```diff
@@ -297,20 +297,32 @@ export default Mixin.create(TargetActionSupport, {
     @private
   */
   keyDown(event) {
-    this.sendAction('key-down', get(this, 'value'), event);
+    sendAction('key-down', this, event);
   },
 });
 
 // In principle, this shouldn't be necessary, but the legacy
 // sendAction semantics for TextField are different from
 // the component semantics so this method normalizes them.
 function sendAction(eventName, view, event) {
-  let action = get(view, `attrs.${eventName}`) || get(view, eventName);
+  let actionName = get(view, `attrs.${eventName}`) || get(view, eventName);
   let value = get(view, 'value');
 
-  view.sendAction(eventName, value);
+  if (typeof actionName === 'string') {
+    deprecate(
+      `Passing actions to components as strings (like {{input ${eventName}="${actionName}"}}) is deprecated. Please use closure actions instead ({{input ${eventName}=(action "${actionName}")}})`,
+      false,
+      { id: 'ember-input.send-action', until: '4.0.0' }
```

Comment:
Note to self: Use the same id than in the `sendAction` deprecation even if the text is different, as in it's essence it's the same feature.